pages/advanced.tsx

```diff
@@ -121,17 +48,15 @@ function Advanced() {
             <option value="3">Ability</option>
             <option value="4">Egg Group</option>
           </Select>
-          <Dropdown
-            typeOfSearch={typeOfSearch}
-            setSearchInput={setSearchInput}
-          />
+          {typeOfSearch == "1" && typeOfDropdown["1"]}
+          {typeOfSearch == "2" && typeOfDropdown["2"]}
+          {typeOfSearch == "3" && typeOfDropdown["3"]}
+          {typeOfSearch == "4" && typeOfDropdown["4"]}
```

Comment:
Can simplify to 
`{['1', '2', '3', '4'].includes(typeOfSearch) && typeOfDropdown[typeOfSearch as '1' | '2' | '3' | '4']}`

To decrease the amount of statements.